### PR TITLE
Generalize scss processing

### PR DESCRIPTION
**Pull request by Bjvanminnen**
Might be easiest to look a commit at a time to see what's changing in the script.

### CODE REVIEW

File: apps/script/convert-scss-variables.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    var isNumber = false;
+    if (/px$/.test(variableValue)) {
+      variableValue = parseFloat(variableValue);
+    }
+    cachedVariables[variableName] = variableValue;
+
+    // Put quotes around non-numbers
+    if (!isNumber) {
+      variableValue = '"' + variableValue + '"';
+    }
```
Comment (Bjvanminnen) [inline apps/script/convert-scss-variables.js:72]: This all works for what we have so far, but I'm a little worried it's not as predictable as we'd like, for example

```
$foo: 13      # becomes { foo: "13" }
$bar: 13px    # becomes { bar: 13 }
```

Could try to generalize/future-proof it better, or we could assume that if/when we expand usage in the future, we deal with it then. Thoughts?

File: apps/Gruntfile.js [hunk 2]
```diff
@@ -288,7 +288,7 @@ module.exports = function (grunt) {
 
   config.exec = {
     browserify: 'echo "' + browserifyExec + '" && ' + browserifyExec,
-    buildColorJs: './script/build-color-js.js',
+    buildColorJs: './script/convert-scss-variables.js',
```
Comment (islemaster) [inline apps/Gruntfile.js]: Should maybe rename the exec task too.

File: apps/script/convert-scss-variables.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+/**
+ * Script that processes an scss file containing a set of variables, and
+ * generates the equivalent set in JS.
+ */
+var fs = require('fs');
+var path = require('path');
+var readline = require('readline');
+
+// Regular expression to capture a color definition from SCSS
```
Comment (islemaster) [inline apps/script/convert-scss-variables.js]: s/color/variable ?

File: apps/script/convert-scss-variables.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    value = cachedVariables[value.substring(1)];
+  }
+  return value;
+}
+
+function convertScssToJs(scssPath, jsPath) {
+  // Generate <foo>.js while reading <foo>.scss line-by-line
+  var out = fs.createWriteStream(jsPath);
+  out.write([
+    '// color.js',
```
Comment (islemaster) [inline apps/script/convert-scss-variables.js]: Should adjust the filename in the header comment according to the `jsPath` argument.

File: apps/script/convert-scss-variables.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    '// GENERATED FILE: DO NOT MODIFY DIRECTLY',
+    '// This generated file exports all colors defined in ' + scssPath,
```
Comment (islemaster) [inline apps/script/convert-scss-variables.js]: s/colors/variables in this comment, too.

File: apps/script/convert-scss-variables.js (hunk 1 above)
Comment (islemaster) [inline apps/script/convert-scss-variables.js:72]: You could generalize a little by changing your isNumber regex to something like this:

``` javascript
/^[\d\.]+(?:px)?$/
```

But for the most part I think we can do what works now, and expect to adapt down the road.

**Comment by islemaster** [comment]
:+1: I think I like this better.  Looks good.

**Pull request merged**